irksome/base_time_stepper.py

```diff
@@ -0,0 +1,155 @@
+import numpy
+
+from abc import abstractmethod
+from firedrake import Function, NonlinearVariationalProblem, NonlinearVariationalSolver
+from firedrake.dmhooks import pop_parent, push_parent
+from firedrake.petsc import PETSc
+from .tools import AI, get_stage_space, getNullspace
+from pyop2.types import MixedDat
+
+
+class BaseTimeStepper:
```

Comment:
I know this is a "developers-only" class, but I still wonder if some documentation on the class and methods would be good to have

Reply:
I just added some doc strings.